roles/ec2-app/tasks/nfs-templates.yml

```diff
@@ -1,30 +1,40 @@
 - block:
   - set_fact:
       nfs_host: "{{ groups['nfs-' + target][0] }}"
+
   - set_fact:
       nfs_launch_time: "{{ hostvars[nfs_host]['create_time'][0:19]|regex_replace('T|Z', ' ') }}"
+      nfs_owner: false
+      nfs_group: false
+      nfs_mode: false
 
-  - set_fact:
-      nfs_owner: "{{ appdata.nfs_template_ownership.owner }}"
-    when:
-      - appdata.nfs_template_ownership.owner is defined
+  - block:
+    - set_fact:
+        nfs_owner: "{{ appdata.nfs_template_ownership.owner }}"
+      when: "'owner' in appdata.nfs_template_ownership"
 
-  - set_fact:
-      nfs_group: "{{ appdata.nfs_template_ownership.group }}"
-    when:
-      - appdata.nfs_template_ownership.group is defined
+    - set_fact:
+        nfs_group: "{{ appdata.nfs_template_ownership.group }}"
+      when: "'group' in appdata.nfs_template_ownership"
```

Comment:
L18, L14: lack of consistency with the other end clauses. 
are you able to drop the begin/end double-quotes, or does the usage of single quotes forces you to wrap it all in double-quotes ?